### ISSUE TITLE
[A11Y][UX] Permissions: Visually indicate that the role column headers include a tooltip (and make sure it's accessible)

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#5989]
This is a follow-up to #5748. As part of that issue, we've included tooltips in the header of permission matrices:
![image](https://user-images.githubusercontent.com/2423362/220075467-7a2a482e-b110-4ff2-a686-a6de0d1d4d0c.png)
(in the screenshot above, I'm hovering my mouse over the "ANONYMOUS" text in the column header)

However, there's a few problems with the implementation (`title` attribute on plain text):
1. The `title` attribute is NOT accessible.
2. There's no visual indication of these tooltips being there.
3. We could add some CSS style as visual indication, that however would only work for sighted people, and only on devices that have a pointer.
4. We could make use of the `aria-describedby` attribute, but that needs to be tied to a focusable element, and plain text does not meet that need by default (see https://stackoverflow.com/questions/1599660/which-html-elements-can-receive-focus for instance).

I have read various relevant articles, but I would like to rehash this specific outline, since it seems to summarize all things I have come across:

> - Only interactive elements should trigger tooltips
> - Tooltips should directly describe the UI control that triggers them (i.e. do not create a control purely to trigger a tooltip)
> - Use `aria-describedby` or `aria-labelledby` to associate the UI control with the tooltip. Avoid `aria-haspopup` and `aria-live`
> - Do not use the `title` attribute to create a tooltip
> - Do not put essential information in tooltips
> - Provide a means to dismiss the tooltip with both keyboard and pointer
> - Allow the mouse to easily move over the tooltip without dismissing it
> - Do not use a timeout to hide the tooltip

## Proposed solution
I have not been able to find a _simple_ solution that isn't bad accessibility-wise, or that doesn't conflict with best UI/UX practices. Articles keep pointing out that we "should be providing the same information to all audiences" and that we "should not use tooltips for important information", which would either force us to remove this feature entirely (as non-accessible), or compromize with a less-than-ideal-solution. I believe that although this may seem unfair, we should still provide this information to "some" people (which is what we are doing currently), since it _is_ useful, instead of not showing it at all.

The other alternative would be to include all that information in the form of plain text, which would overwhelm the form, and for which we would have no other option but to place it out of context (for example at the top of the page as a huge chunk of help text). That approach might work well for help text that describes the entire page (like the collapsed help text we have at the top of the layouts listing page), but what we are trying to describe here is specific, individual items - each informational text excerpt needs to be tied to the element it provides clarification/details for. Otherwise, there's no point 🤷🏼 

At this point, I would like to mention https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role
> A `tooltip` is a contextual text bubble that displays a description for an element that appears on pointer hover or keyboard focus. ... Tooltips provide contextual information about an element when that owning element receives focus or is hovered over, but is otherwise not visible on the page. ...

So by definition, tooltips are to be used with a pointing device or a keyboard.

Anyway, bottom line is that I would like us to:
1. Consider the following 2 options:
   - adding `tabindex=0` to the text, to make it focusable. The practice of adding `tabindex` to non-interactive elements is normally advised against, but in this specific scenario I believe that we are playing by the rules, since the element _is_ interactive (it presents a tooltip with more information)
   - if the previous option is not acceptable, then we should make the text a link (but style it differently - as plain text with a dotted underline and/or a `cursor: help;`).
2. Do not use the `title` attribute, since that is not accessible.
3. Use the `aria-describedby` attribute instead, to link the text to its description.

## *Resources (there are so many on this topic out there)
- https://www.a11yproject.com/posts/title-attributes
- https://www.tpgi.com/html5-accessibility-chops-title-attribute-use-and-abuse
- https://silktide.com/blog/i-thought-title-text-improved-accessibility-i-was-wrong
- https://sarahmhigley.com/writing/tooltips-in-wcag-21/#best-practices-summary
- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role
- https://www.w3.org/WAI/ARIA/apg/patterns/tooltip
- https://inclusive-components.design/tooltips-toggletips (this is interesting, as it introduces the notion of "toggletips" as a different form of tooltip)
- https://www.w3.org/TR/WCAG20-TECHS/ARIA1.html
- https://www.a11yproject.com/posts/how-to-use-the-tabindex-attribute/#making-non-interactive-elements-focusable
